Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp

```diff
@@ -138,32 +137,44 @@ namespace AZ::SceneAPI::Behaviors
                         // get the MeshData child node index values for Transform and CustomPropertyData
                         auto childIndex = it.GetHierarchyIterator()->GetChildIndex();
 
-                        MeshNodeData meshNodeData;
-                        meshNodeData.m_meshIndex = currentIndex;
+                        NodeDataForEntity nodeDataForEntity;
+                        nodeDataForEntity.m_meshIndex = currentIndex;
 
                         while (childIndex.IsValid())
                         {
                             const auto childContent = graph.GetNodeContent(childIndex);
-                            if (currentContent.get())
+                            if (childContent)
                             {
                                 if (azrtti_istypeof<AZ::SceneAPI::DataTypes::ITransform>(childContent.get()))
                                 {
-                                    meshNodeData.m_transformIndex = childIndex;
+                                    nodeDataForEntity.m_transformIndex = childIndex;
                                 }
                                 else if (azrtti_istypeof<AZ::SceneAPI::DataTypes::ICustomPropertyData>(childContent.get()))
                                 {
-                                    meshNodeData.m_propertyMapIndex = childIndex;
+                                    nodeDataForEntity.m_propertyMapIndex = childIndex;
                                 }
                             }
                             childIndex = graph.GetNodeSibling(childIndex);
                         }
 
-                        meshDataMap.emplace(MeshDataMapEntry{ currentIndex, AZStd::move(meshNodeData) });
+                        nodeDataMap.emplace(NodeDataMapEntry{ currentIndex, AZStd::move(nodeDataForEntity) });
+                    }
+                    else if (azrtti_istypeof<AZ::SceneAPI::DataTypes::ITransform>(currentContent.get()))
+                    {
+                        // Check if this transform node is not associated with any meshes
+                        auto parentNodeIndex = graph.GetNodeParent(currentIndex);
+                        const auto parentContent = graph.GetNodeContent(parentNodeIndex);
+                        if (!azrtti_istypeof<AZ::SceneAPI::DataTypes::IMeshData>(parentContent.get()))
+                        {
+                            NodeDataForEntity nodeDataForEntity;
+                            nodeDataForEntity.m_transformIndex = currentIndex;
+                            nodeDataMap.emplace(NodeDataMapEntry{ currentIndex, AZStd::move(nodeDataForEntity) });
```

Comment:
Will this create an extra entity for the graphs that have nodes for both mesh data and transform data? 
That is, will there now be duplicate entities for scene graphs that have Mesh Data nodes?
If so, this will lead to bloated prefabs.

Reply:
No, it shouldn't create duplicates because the above lines will only get triggered if the parent of the transform node is not a mesh node.